File: Leetcode/1557 Minimum Number of Vertices to Reach All Nodes.ts
function findSmallestSetOfVertices(n: number, edges: number[][]): number[] {
    const inDegree = new Array<number>(n).fill(0);
    for (const [_, v] of edges) {
        inDegree[v]++;
    }
    return inDegree.reduce(
        (minCover, inDeg, node) => {
            if (inDeg === 0) minCover.push(node);
            return minCover;
        },
        new Array<number>()
    );
}

File: Leetcode/1557 Minimum Number of Vertices to Reach All Nodes.py
class Solution:
    def findSmallestSetOfVertices(self, n, edges):
        in_degree = [0] * n
        for _, v in edges:
            in_degree[v] += 1
        return [node for node, in_deg in enumerate(in_degree) if in_deg == 0]
 
# without list comprehension
 class Solution:
    def findSmallestSetOfVertices(self, n, edges):
        in_degree = [0] * n
        for _, v in edges:
            in_degree[v] += 1
        min_cover = []
        for node, in_deg in enumerate(in_degree):
            if in_deg == 0:
                min_cover.append(node)
        return min_cover
